OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -2051,6 +2051,32 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				if (engineVersion < 20150902)
+				{
+					// Replace RenderSimple with RenderSprites + WithSpriteBody + AutoSelectionSize
+					if (depth == 0)
+					{
+						var rs = node.Value.Nodes.FirstOrDefault(n => n.Key.StartsWith("RenderSimple"));
+						if (rs != null)
+						{
+							rs.Key = "RenderSprites";
+
+							var wsbNodes = rs.Value.Nodes.Where(n => n.Key == "Sequence").ToList();
+							if (wsbNodes.Any())
+								node.Value.Nodes.Add(new MiniYamlNode("WithSpriteBody", new MiniYaml("", wsbNodes)));
+							else
+								node.Value.Nodes.Add(new MiniYamlNode("WithSpriteBody", ""));
+
+							node.Value.Nodes.Add(new MiniYamlNode("AutoSelectionSize", ""));
+							rs.Value.Nodes.RemoveAll(n => wsbNodes.Contains(n));
+						}
+
+						var rrs = node.Value.Nodes.FirstOrDefault(n => n.Key.StartsWith("-RenderSimple"));
+						if (rrs != null)
+							rrs.Key = "-RenderSprites";
```

Comment:
This will generate bogus rules for automatically upgraded actors.  `WithSpriteBodyInfo` `Requires<RenderSpritesInfo>`.